### DISCUSSION TITLE
Use ChatGPT to generate potential domain names

### DISCUSSION THREAD

**Discussion by sullo** [projectdiscovery/subfinder#1127] [Ideas]
ChatGPT (or other AI) is capable of identifying patterns in domain names and generating new names based on the patterns. As a data source, this could be used to find additional hostnames.

### Please describe your feature request:
Create a precise prompt to generate new names based on a list of existing names, and integrate with subfinder for additional name discovery.

Example:
```
Using the rules defined in RFC1034 and the other RFCs which impact fully qualified domain names (FQDN), generate a list of 20 more potential DNS names based on any observable patterns based on the following list. Do not explain. 

admin.acrobat.com
admin.adobe.acrobat.com
admin.adobe.stage.acrobat.com
admin.beta.acrobat.com
admin.beta.emea.acrobat.com
admin.connectpro.acrobat.com
admin.emea.acrobat.com
admin.emea.adobe.acrobat.com
admin.emea.stage.acrobat.com
admin.na3.acrobat.com
admin.na4.acrobat.com
admin.na5.acrobat.com
admin.na6.acrobat.com
comments.r53.acrobat.com
compete.na3.acrobat.com
conferencecasting.na5.acrobat.com
```

Response:
```
accounts.na3.acrobat.com
analytics.na4.acrobat.com
billing.emea.acrobat.com
billing.na6.acrobat.com
careers.adobe.acrobat.com
customersupport.na5.acrobat.com
developer.beta.acrobat.com
developer.emea.acrobat.com
devtools.na4.acrobat.com
download.adobe.acrobat.com
events.na6.acrobat.com
helpdesk.na3.acrobat.com
marketing.beta.acrobat.com
marketingconnect.na5.acrobat.com
myaccount.adobe.acrobat.com
partners.emea.acrobat.com
partners.na4.acrobat.com
payment.emea.acrobat.com
press.na5.acrobat.com
webinar.connectpro.acrobat.com
```

### Describe the use case of this feature:
Find additional names.

**Comment by sullo** [comment]
Note that similar functionality can be achieved w/o AI:
https://cramppet.github.io/regulator/index.html

**Comment by tarunKoyalwar** [comment]
> Note that similar functionality can be achieved w/o AI: https://cramppet.github.io/regulator/index.html

I think this feature w/o (i.e without ai) should be implemented in `dnsx` since this would contain lot of false positive and we would have to check/validate if subdomain is alive (which is active subdomain enum).

The case with chatgpt is same but I think we can allow this since chatgpt has more context and generate finite results

**Comment by tarunKoyalwar** [comment]
-generating subs without AI (ref: https://github.com/jfrancois/SDBF) It is done with character level analysis

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
